include/constants/trainer_hill.h

```diff
@@ -1,27 +1,46 @@
 #ifndef GUARD_CONSTANTS_TRAINER_HILL_H
 #define GUARD_CONSTANTS_TRAINER_HILL_H
 
+#define TRAINER_HILL_1F        1
+#define TRAINER_HILL_2F        2
+#define TRAINER_HILL_3F        3
+#define TRAINER_HILL_4F        4
+#define TRAINER_HILL_ROOF      5
+#define TRAINER_HILL_ENTRANCE  6
+
+#define NUM_TRAINER_HILL_FLOORS    4
+#define NUM_TRAINER_HILL_FLOORS_JP 2
+
+#define NUM_TRAINER_HILL_PRIZE_LISTS 10
+
 #define TRAINER_HILL_FUNC_START 0
-#define TRAINER_HILL_FUNC_1 1
-#define TRAINER_HILL_FUNC_2 2
-#define TRAINER_HILL_FUNC_3 3
+#define TRAINER_HILL_FUNC_GET_OWNER_STATE 1
+#define TRAINER_HILL_FUNC_GIVE_PRIZE 2
+#define TRAINER_HILL_FUNC_CHECK_FINAL_TIME 3
 #define TRAINER_HILL_FUNC_RESUME_TIMER 4
 #define TRAINER_HILL_FUNC_SET_LOST 5
 #define TRAINER_HILL_FUNC_GET_CHALLENGE_STATUS 6
-#define TRAINER_HILL_FUNC_7 7
-#define TRAINER_HILL_FUNC_8 8
-#define TRAINER_HILL_FUNC_9 9
-#define TRAINER_HILL_FUNC_IS_IN 10
-#define TRAINER_HILL_FUNC_11 11
-#define TRAINER_HILL_FUNC_12 12
-#define TRAINER_HILL_FUNC_13 13
-#define TRAINER_HILL_FUNC_14 14
-#define TRAINER_HILL_FUNC_15 15
-#define TRAINER_HILL_FUNC_16 16
+#define TRAINER_HILL_FUNC_GET_CHALLENGE_TIME 7
+#define TRAINER_HILL_FUNC_GET_ALL_FLOORS_USED 8
+#define TRAINER_HILL_FUNC_CLEAR_RESULT 9
+#define TRAINER_HILL_FUNC_IN_CHALLENGE 10
+#define TRAINER_HILL_FUNC_POST_BATTLE_TEXT 11
+#define TRAINER_HILL_FUNC_SET_ALL_TRAINER_FLAGS 12
+#define TRAINER_HILL_FUNC_GET_GAME_SAVED 13
+#define TRAINER_HILL_FUNC_SET_GAME_SAVED 14
+#define TRAINER_HILL_FUNC_CLEAR_GAME_SAVED 15
+#define TRAINER_HILL_FUNC_GET_WON 16
 #define TRAINER_HILL_FUNC_SET_TAG 17
 
-// Values returned by the TrainerHillGetChallengeStatus
-// function.
+#define TRAINER_HILL_TEXT_INTRO        2
+#define TRAINER_HILL_TEXT_PLAYER_LOST  3
+#define TRAINER_HILL_TEXT_PLAYER_WON   4
+#define TRAINER_HILL_TEXT_AFTER        5
+
+#define NUM_TRAINER_HILL_TRAINERS     (NUM_TRAINER_HILL_FLOORS * 2)
```

Comment:
Is this relationship definitely supposed to be `num_floors * 2`, or does it just happen to be that for the existing data in the game?

Reply:
Yeah there's always 2 trainers per floor. Even when E-Reader trainers are used, if there's an odd number of trainers then one is added to round it up.